activesupport/test/xml_mini/xml_mini_engine_test.rb

```diff
@@ -0,0 +1,260 @@
+require "abstract_unit"
+require "active_support/xml_mini"
+require "active_support/core_ext/hash/conversions"
+
+class XMLMiniEngineTest < ActiveSupport::TestCase
+  def self.run_with_gem(gem_name)
+    require gem_name
+    yield
+  rescue LoadError
+    # Skip tests unless gem is available
+  end
+
+  def self.run_with_platform(platform_name)
+    yielf if RUBY_PLATFORM.include?(platform_name)
+  end
+
+  def self.inherited(base)
+    base.class_eval do
+      include EngineTests
+    end
+
+    super
+  end
+
+  module EngineTests
+    def setup
+      @default_backend = ActiveSupport::XmlMini.backend
+      ActiveSupport::XmlMini.backend = engine
+      super
+    end
+
+    def teardown
+      ActiveSupport::XmlMini.backend = @default_backend
+      super
+    end
```

Comment:
`setup` and `teardown` can live in the base class just fine. Then I think we can also strip the `ActiveSupport::` namespace like they were in the original test files.